core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -2895,7 +2896,31 @@ private RecordCursorFactory generateSelectChoose(QueryModel model, SqlExecutionC
                         )
                 );
             } else {
-                selectMetadata.add(colMetadata);
+                if (selectMetadata.getColumnIndexQuiet(colMetadata.getName()) < 0) {
+                    selectMetadata.add(colMetadata);
+                } else {
+                    // avoid clashing with other columns using timestamp column name as alias
+                    StringSink sink = Misc.getThreadLocalSink();
+                    sink.put(colMetadata.getName());
+                    int len = sink.length();
+                    int sequence = 0;
+
+                    do {
+                        sink.trimTo(len);
+                        sink.put(sequence++);
+                    } while (selectMetadata.getColumnIndexQuiet(colMetadata.getName()) < 0);
```

Comment:
Should it be instead?
```suggestion
                    } while (selectMetadata.getColumnIndexQuiet(sink) > -1);
```

I don't understand how this code is covered by tests...

Reply:
You're right. I forgot to change this expression and test worked because there was no column to clash with 'timestamp0' .
Thanks for catching this !
Fixed in latest commit.